library/alloc/src/collections/btree/node.rs

```diff
@@ -1003,7 +1006,10 @@ impl<BorrowType: marker::BorrowType, K, V>
     /// `edge.descend().ascend().unwrap()` and `node.ascend().unwrap().descend()` should
     /// both, upon success, do nothing.
     pub fn descend(self) -> NodeRef<BorrowType, K, V, marker::LeafOrInternal> {
-        let _ = BorrowType::TRAVERSAL_PERMIT;
+        const {
+            assert!(BorrowType::TRAVERSAL_PERMIT);
+        }; // #[cfg(bootstrap)] remove ;
```

Comment:
why should this be removed in the next cycle?

Reply:
Inline const parses as expression with block after https://github.com/rust-lang/rust/pull/105142

Reply:
Oh, apparently this is in beta already and just blocked on https://github.com/rust-lang/rust/pull/105741

Reply:
Ok, let's wait until that gets merged so you can rebase over it and avoid needing cfg(bootstrap).